process/block/shardblock_test.go

```diff
@@ -1367,6 +1263,303 @@ func TestShardProcessor_ProcessBlockWithWrongMiniBlockHeaderShouldErr(t *testing
 	assert.Equal(t, process.ErrHeaderBodyMismatch, err)
 }
 
+// ------- requestEpochStartInfo
+func TestShardProcessor_RequestEpochStartInfo(t *testing.T) {
+	t.Parallel()
+
+	coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
+
+	headerHash := []byte("hash")
+	headerEpoch := uint32(101)
+	blockStartOfEpoch := &block.Header{
+		Epoch:              headerEpoch,
+		EpochStartMetaHash: headerHash,
+	}
+	errGetHeader := errors.New("error getting headers")
+
+	t.Run("header is not start of epoch, should not request any header", func(t *testing.T) {
+		t.Parallel()
+
+		args := CreateMockArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+		args.RequestHandler = &testscommon.RequestHandlerStub{
+			RequestMetaHeaderCalled: func(hash []byte) {
+				require.Fail(t, "should not try to request meta header")
+			},
+		}
+
+		shardProc, _ := blproc.NewShardProcessor(args)
+		header := &testscommon.HeaderHandlerStub{
+			IsStartOfEpochBlockCalled: func() bool {
+				return false
+			},
+		}
+		err := shardProc.RequestEpochStartInfo(header, haveTime)
+		require.Nil(t, err)
```

Comment:
it is mandatory to add a time.Sleep here of about 1 second in order to be sure no request is made
Reason: the request is done on a go routine that will need some time to get executed.
Valid on all tests like this

Reply:
Right, missed this part. Added a `time.Sleep(time.Second)` for all tests to be sure